polkadot/node/network/collator-protocol/src/validator_side/tests/prospective_parachains.rs

```diff
@@ -1311,3 +1436,646 @@ fn child_blocked_from_seconding_by_parent(#[case] valid_parent: bool) {
 		virtual_overseer
 	});
 }
+
+#[test]
+fn collations_outside_limits_are_not_fetched() {
+	let test_state = TestState::with_shared_core();
+
+	test_harness(ReputationAggregator::new(|_| true), |test_harness| async move {
+		let TestHarness { mut virtual_overseer, keystore } = test_harness;
+
+		let head_b = Hash::from_low_u64_be(128);
+		let head_b_num: u32 = 2;
+
+		update_view(
+			&mut virtual_overseer,
+			&test_state,
+			vec![(head_b, head_b_num)],
+			1,
+			&test_state.async_backing_params,
+		)
+		.await;
+
+		let peer_a = PeerId::random();
+		let pair_a = CollatorPair::generate().0;
+
+		connect_and_declare_collator(
+			&mut virtual_overseer,
+			peer_a,
+			pair_a.clone(),
+			test_state.chain_ids[0],
```

Comment:
shouldn't we use the same `TestState::CHAIN_IDS[0]` as below?

Reply:
They are the same values but for consistency I've replaced `TestState::CHAIN_IDS` with `test_state.chain_ids` everywhere.